### PR TITLE
[backend] Fix on get exercise details to have better performance

### PR DESCRIPTION
**Pull request by Dimfacion**
### Proposed changes

* Fix on get exercise details to have better performance
*

### Related issues

*
*

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by Christian-DONGMO — CHANGES_REQUESTED** [review]

File: openbas-api/src/main/java/io/openbas/rest/exercise/form/ExerciseDetails.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  @JsonProperty("exercise_name")
+  @NotBlank
+  private String name;
+
+  @JsonProperty("exercise_description")
+  private String description;
+
+  @JsonProperty("exercise_status")
+  @Enumerated(EnumType.STRING)
+  private Exercise.STATUS status;
```
[the remark this answers is not in the document]
Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/exercise/form/ExerciseDetails.java]: Makes sense :) Done 👍

File: openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseApi.java
[the remark this answers is not in the document]
Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseApi.java:1]: First point is done. Thanks for bringing this up :)

As for the second one, while I do agree on the idea, I don't think this should be the focus of the team right now and especially not in this fix. I'm okay with small improvments like the use of lombok but not with architectural changes in fixes.

**Review by Christian-DONGMO — APPROVED** [review]

File: openbas-api/src/main/java/io/openbas/rest/exercise/form/ExerciseDetails.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  @JsonProperty("exercise_message_footer")
+  private String footer;
+
+  @JsonProperty("exercise_mail_from")
+  private String from;
+
+  @JsonProperty("exercise_mails_reply_to")
+  private List<String> replyTo;
+
+  @JsonProperty("exercise_logo_dark")
```
[the remark this answers is not in the document]
Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/exercise/form/ExerciseDetails.java]: Removed

File: openbas-api/src/main/java/io/openbas/rest/exercise/form/ExerciseDetails.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+  private String logoDark;
+
+  @JsonProperty("exercise_logo_light")
```
[the remark this answers is not in the document]
Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/exercise/form/ExerciseDetails.java]: Removed

File: openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseApi.java [hunk 4]
```diff
@@ -556,8 +411,100 @@ public void deleteExercise(@PathVariable String exerciseId) {
 
   @GetMapping("/api/exercises/{exerciseId}")
   @PreAuthorize("isExerciseObserver(#exerciseId)")
```
[the remark this answers is not in the document]
Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseApi.java:413]: Done

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**